File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.destination.bigquery;

import com.google.cloud.bigquery.BigQueryException;
import io.airbyte.cdk.integrations.base.AirbyteExceptionHandler;
import io.airbyte.cdk.integrations.base.IntegrationCommand;
import io.airbyte.cdk.integrations.base.context.AirbyteConnectorRunner;

public class BigQueryDestination {

  public static void main(final String[] args) {
    AirbyteExceptionHandler.addThrowableForDeinterpolation(BigQueryException.class);
    AirbyteConnectorRunner.run(IntegrationCommand.class, args);
  }

}

File: airbyte-integrations/connectors/destination-bigquery/src/main/kotlin/io/airbyte/integrations/destination/bigquery/BigQueryDestination.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.integrations.destination.bigquery

import io.airbyte.cdk.AirbyteDestinationRunner
import io.airbyte.cdk.load.command.aws.AwsToolkitConstants

/**
 * This is needed because the GCS client is, under the hood, using the S3Client.
 *
 * And the S3Client depends on the AWS environment - we're not actually _using the assume role
 * stuff, but the wiring needs to be satisfied.
 */
val additionalMicronautEnvs = listOf(AwsToolkitConstants.MICRONAUT_ENVIRONMENT)

fun main(args: Array<String>) {
    AirbyteDestinationRunner.run(*args, additionalMicronautEnvs = additionalMicronautEnvs)
}
